trieste/models/gpflow/models.py

```diff
@@ -260,58 +265,58 @@ def conditional_predict_f(
+        cov_shape = tf.shape(cov_add)
         noise = self.model.likelihood.variance * tf.eye(
-            cov_shape[-2], batch_shape=cov_shape[:-2], dtype=cov_old.dtype
+            cov_shape[-2], batch_shape=cov_shape[:-2], dtype=cov_add.dtype
         )
-        L_old = tf.linalg.cholesky(cov_old + noise)  # [..., L, N, N]
-        A = tf.linalg.triangular_solve(L_old, cov_cross, lower=True)  # [..., L, N, M]
-        var_new = var_new - leading_transpose(
+        L_add = tf.linalg.cholesky(cov_add + noise)  # [..., L, N, N]
+        A = tf.linalg.triangular_solve(L_add, cov_cross, lower=True)  # [..., L, N, M]
+        var_qp_new = var_qp - leading_transpose(
             tf.reduce_sum(A ** 2, axis=-2), [..., -1, -2]
-        )  # [M, L]
+        )  # [..., M, L]
 
-        mean_old_diff = additional_data.observations - mean_old  # [..., N, L]
-        mean_old_diff = leading_transpose(mean_old_diff, [..., -1, -2])[..., None]  # [..., L, N, 1]
-        AM = tf.linalg.triangular_solve(L_old, mean_old_diff)  # [..., L, N, 1]
+        mean_add_diff = additional_data.observations - mean_add  # [..., N, L]
+        mean_add_diff = leading_transpose(mean_add_diff, [..., -1, -2])[..., None]  # [..., L, N, 1]
+        AM = tf.linalg.triangular_solve(L_add, mean_add_diff)  # [..., L, N, 1]
 
-        mean_new = mean_new + leading_transpose(
+        mean_qp_new = mean_qp + leading_transpose(
             (tf.matmul(A, AM, transpose_a=True)[..., 0]), [..., -1, -2]
         )  # [..., M, L]
 
         tf.debugging.assert_shapes(
             [
-                (additional_data.query_points, [..., "N", "D"]),
                 (additional_data.observations, [..., "N", "L"]),
                 (query_points, ["M", "D"]),
-                (mean_new, [..., "M", "L"]),
-                (var_new, [..., "M", "L"]),
+                (mean_qp_new, [..., "M", "L"]),
+                (var_qp_new, [..., "M", "L"]),
             ],
             message="received unexpected shapes computing conditional_predict_f,"
             "check model kernel structure?",
         )
 
-        return mean_new, var_new
+        return mean_qp_new, var_qp_new
 
     def conditional_predict_joint(
         self, query_points: TensorType, additional_data: Dataset
     ) -> tuple[TensorType, TensorType]:
         """
         Predicts the joint GP distribution at query_points conditioned on both the model
-        and some additional data, using exact formula.
+        and some additional data, using exact formula. See :cite:`chevalier2014corrected`
+        (eqs. 8-10) for details.
 
         :param query_points: Set of query points with shape [M, D]
         :param additional_data: Dataset with query_points with shape [..., N, D] and observations
                  with shape [..., N, L]
-        :return: mean_new: predictive variance at query_points, with shape [..., M, L],
+        :return: mean_qp: predictive variance at query_points, with shape [..., M, L],
```

Comment:
inconsistency with `cov_new` below is here for a reason? also, here it says "variance", it should be mean, right?